### ISSUE TITLE
In-game chat always keeps focus, prevents custom shortcuts from working.

### ISSUE THREAD

**Issue by chtref** [Cockatrice/Cockatrice#1902]
<b>OS:</b>
Windows 7 Service Pack 1

<b>Cockatrice version:</b>
f217551

---

<br>

Most of the custom in-game shortcuts like " \* for Set Annotation" or "Numpad7 for Set Counters Green" no longer work because of the chat grabbing the focus at all times (which mean you always see \* and 7 in the chat in my example instead of the proper actions).

**Comment by ctrlaltca** [comment]
> the chat grabbing the focus at all times

We have two focus area: the game and the chat.
If the chat is focused, it will grab "7" and "*" as characters you may want to type on the chat.
You should be able to click on the game area and have the shortcuts working.
If the chat steals the focus by itself, instead, this is a bug; in this case, are you able to tell us how to reproduce this problem?

**Comment by chtref** [comment]
The chat steals the focus by itself.

The way I tested it is the following:

1. Click on a card on the battlefield. It will be selected with a red border.
2. Look at the chat input textfield. It will still have the focus. If you try using your shortcut, it will type the command instead. Look at the OP for an example with * → Set Annotation.

**Comment by chtref** [comment]
I have gatherered more information of the behavior of the chat input textfield's focus tendancies.

Basically, it is possible for the chatbox input textfield to lose focus if you click on the chatbox output. Unfortunately, it is the only way to do it: the game area does not remove focus from the chatbox input textfield. Shortcuts will work if you remove the focus from the chatbox input textfield, which means clicking on the chatbox output is a temporary solution.

**Comment by tooomm** [comment]
Turns out to be not as easy when thinking about the best/right way on how/when to swap focus and I couldn't come up with the perfect way to handle it right now...

But in any case...
The app should differentiate between a game that is running and a waiting/finished game lobby though. If you just open a new game and are waiting for players to join or your opponent to finish sideboarding between games you want the chat box to always have focus instead the game area.
The app should also switch focus on game start once to focus the game area.

**Comment by chtref** [comment]
Let's assume my shortcut is "Numpad4: Add 1 green counter".

The chatbox output is that chat itself. The log, basically.

When you launch the game, the focus is on the game area. When you click on the chatbox input, it's gonna keep the focus until you click on something else that can remove the focus from the chat. In my example, I use the chatbox output, aka the chat log.

![cockatrice_focuschat](https://cloud.githubusercontent.com/assets/10520926/14340509/72ae3e48-fc56-11e5-9c59-0eb14d480d18.gif)

In my example, I try to add counters via my shortcut. I then give the focus to the chatbox input and try again. I then click on the chatbox output and try again one final time.

Basically, the big problem is that the chatbox input will not unfocus when I click on the game area.

**Comment by tooomm** [comment]
chatbox output = chat log, ok :+1:

**Comment by ZeldaZach** [comment]
Fixed, you can now press escape to unfocus
